File: app/src/main/java/com/example/groxweatherapp/grox/WeatherRefreshAction.java
package com.example.groxweatherapp.grox;

import com.example.groxweatherapp.model.WeatherModel;
import com.groupon.grox.Action;

import static com.example.groxweatherapp.model.WeatherModel.WeatherModelState.REFRESHING;

public class WeatherRefreshAction implements Action<WeatherModel> {

    public WeatherRefreshAction() {
        // no ops
    }

    @Override
    public WeatherModel newState(WeatherModel oldState) {
        return oldState.toBuilder()
            .setModelState(REFRESHING)
            .build();
    }
}

File: app/src/main/java/com/example/groxweatherapp/grox/WeatherRefreshAction.kt
package com.example.groxweatherapp.grox

import com.example.groxweatherapp.model.WeatherModel
import com.example.groxweatherapp.model.WeatherModel.Companion.WeatherModelState.REFRESHING
import com.groupon.grox.Action

class WeatherRefreshAction : Action<WeatherModel> {

    override fun newState(oldState: WeatherModel): WeatherModel {
        return oldState.copy(state = REFRESHING)
    }
}
